aten/src/ATen/native/native_functions.yaml

```diff
@@ -6728,6 +6728,12 @@
     SparseCsrCPU, SparseCsrCUDA: sparse_mask_sparse_csr
   autogen: sparse_mask.out
 
+- func: _sparse_mask_projection(Tensor self, Tensor mask) -> Tensor
```

Comment:
Are we sure this needs another native function or can this live in https://github.com/pytorch/pytorch/blob/master/torch/csrc/autograd/FunctionsManual.h/.cpp?

Reply:
@albanD - do you have some requirements here for this / can you take a look?

Reply:
The kernel uses dispatch stabs under the hood, hence the need for the native exterior.

Reply:
We would need more than a single kw argument for that, see https://github.com/pytorch/pytorch/pull/94991#discussion_r1172897761, i.e., the difference is in the direction of projection and/or whether intersecting matches require accumulation or not.